### DISCUSSION TITLE
Versioned Documentation

### DISCUSSION THREAD

**Discussion by tomirgang** [Elektrobit/ebcl_template#79] [Ideas]
Hi all,

I guess it's time to start versioned documentation, as described in https://blog.esciencecenter.nl/versioned-documentation-using-only-github-actions-and-github-pages-1825296e31aa .
Maybe we should also change from mdbook to another documentation engine, to e.g. allow easy creation of PDF versions.

@antonjahn  @simone-weiss  @wintron04 @matthiasb85  What do you think? Do you have proposals for a better documentation engine?

**Comment by antonjahn** [comment]
Personal opinion:
* AsciiDoc + Antora supports versioned docs, primarily served online, PDF export is not primary target, but should also be possible
* Sphinx + readthedocs supports versioned docs, and export to other formats (bonus: sphinx-needs might be useful)

Decision criteria:
* Do we want to integrate our docs with other existing docs (strength of Antora, if they are in adoc)
* Do we want to use something that people are used to (strength of Sphinx/Readthedocs)
* Do we align with other projects and open initiatives?

Are we talking for a single repo only (personal preference sphinx), or do we foresee multiple repos/projects, that we want to document on a single documentation site (personal preference antora)

**Reply by tomirgang** [reply]
Antora looks great on a very first glance.

**Comment by MofX** [comment]
Can the two topics be separated? I think having a version documentation is more important right now than changing the documentation format.

**Comment by antonjahn** [comment]
I would propose to close this discussion with: https://github.com/Elektrobit/ebcl_template/pull/109
The PR adds a simple, extendable version handling without a major rewrite & structure change.
